contracts/src/Teleporter/upgrades/tests/TeleporterRegistryTests.t.sol

```diff
@@ -377,6 +435,71 @@ contract TeleporterRegistryTest is Test {
         teleporterRegistry.addProtocolVersion(messageIndex);
     }
 
+    function testMaxVersionIncrement() public {
+        uint256 latestVersion = teleporterRegistry.latestVersion();
+        uint32 messageIndex = 0;
+
+        // Adding a version that matches the max version increment succeeds
+        WarpMessage memory warpMessage = _createWarpOffChainMessage(
+            latestVersion + teleporterRegistry.MAX_VERSION_INCREMENT(),
+            address(teleporterRegistry),
+            teleporterAddress,
+            address(teleporterRegistry)
+        );
+
+        vm.mockCall(
+            WARP_PRECOMPILE_ADDRESS,
+            abi.encodeCall(
+                IWarpMessenger.getVerifiedWarpMessage,
+                (messageIndex)
+            ),
+            abi.encode(warpMessage, true)
+        );
+        vm.expectCall(
+            WARP_PRECOMPILE_ADDRESS,
+            abi.encodeCall(
+                IWarpMessenger.getVerifiedWarpMessage,
+                (messageIndex)
+            )
+        );
```

Comment:
This `mockCall` and then `expectCall` pattern looks to be similar in a few places. If it's easy, may want to consider making it a helper function, but no strong preference either way.